horovod/spark/common/store.py

```diff
@@ -0,0 +1,338 @@
+
+    In order to allow for caching but to prevent overuse of disk space on intermediate data, intermediate datasets
+    are named in a deterministic sequence. When a dataset is done being used for training, the intermediate files
+    can be reclaimed to free up disk space, but will not be automatically removed so that they can be reused as
+    needed. This is to support both parallel training processes using the same store on multiple DataFrames, as well
+    as iterative training using the same DataFrame on different model variations.
+    """
+    def __init__(self):
+        self._train_data_to_key = {}
+        self._val_data_to_key = {}
+
+    def get_parquet_dataset(self, path):
+        raise NotImplementedError()
+
+    def get_train_data_path(self, idx=None):
+        raise NotImplementedError()
+
+    def get_val_data_path(self, idx=None):
+        raise NotImplementedError()
+
+    def get_test_data_path(self, idx=None):
+        raise NotImplementedError()
+
+    def saving_runs(self):
+        raise NotImplementedError()
+
+    def get_runs_path(self):
+        raise NotImplementedError()
+
+    def get_run_path(self, run_id):
+        raise NotImplementedError()
+
+    def get_checkpoint_path(self, run_id):
+        raise NotImplementedError()
+
+    def get_logs_path(self, run_id):
+        raise NotImplementedError()
+
+    def get_checkpoint_filename(self):
+        raise NotImplementedError()
+
+    def get_logs_subdir(self):
+        raise NotImplementedError()
+
+    def exists(self, path):
+        raise NotImplementedError()
+
+    def read(self, path):
+        raise NotImplementedError()
+
+    def get_local_output_dir_fn(self, run_id):
+        raise NotImplementedError()
+
+    def sync_fn(self, run_id):
+        raise NotImplementedError()
+
+    def to_remote(self, run_id, dataset_idx):
+        attrs = self._remote_attrs(run_id, dataset_idx)
+
+        class RemoteStore(object):
+            def __init__(self):
+                for name, attr in attrs.items():
+                    setattr(self, name, attr)
+
+        return RemoteStore()
+
+    def _remote_attrs(self, run_id, dataset_idx):
+        return {
+            'train_data_path': self.get_train_data_path(dataset_idx),
+            'val_data_path': self.get_val_data_path(dataset_idx),
+            'test_data_path': self.get_test_data_path(dataset_idx),
+            'saving_runs': self.saving_runs(),
+            'runs_path': self.get_runs_path(),
+            'run_path': self.get_run_path(run_id),
+            'checkpoint_path': self.get_checkpoint_path(run_id),
+            'logs_path': self.get_logs_path(run_id),
+            'checkpoint_filename': self.get_checkpoint_filename(),
+            'logs_subdir': self.get_logs_subdir(),
+            'get_local_output_dir': self.get_local_output_dir_fn(run_id),
+            'sync': self.sync_fn(run_id)
+        }
+
+    @staticmethod
+    def create(prefix_path):
```

Comment:
we should match this arguments list as HDFSStore.__ini__

Reply:
Added `*args` and `**kwargs` to match all arguments.